src/composables/layer-metadata/layer-metadata.composable.ts

```diff
@@ -0,0 +1,162 @@
+import { useTranslation } from 'i18next-vue'
+import { ThemeNodeModel } from '../themes/themes.model'
+import { LayerTreeNodeModel } from '@/components/layer-tree/layer-tree.model'
+import { IdValues, LayerMetadataModel } from './layer-metadata.model'
+import {
+  getMetadataLinks,
+  getResponsibleParty,
+  isoLang2To3,
+  stringToHtml,
+} from './layer-metadata.utils'
+import useWmsHelper from '../common/wms-helper.composable'
+import useThemes from '../themes/themes.composable'
+
+export default function useLayerMetadata() {
+  // TODO: get urls from a config
+  const geonetworkBaseUrl = 'https://geocatalogue.geoportail.lu/geonetwork/srv'
+  const legendBaseUrl = 'https://map.geoportail.lu/legends/get_html'
+  // TODO: get from config or relative
+  const localMetadataBaseUrl = 'https://map.geoportail.lu/getMetadata'
+  const themes = useThemes()
+  const wmsHelper = useWmsHelper()
+  const { i18next } = useTranslation()
+
+  async function getLayerMetadata(
+    node: LayerTreeNodeModel
+  ): Promise<LayerMetadataModel> {
+    const layer: ThemeNodeModel | undefined = themes.findById(
+      parseInt(node.id, 10)
+    )
+    let localMetadata
+    let metadata
+    if (layer) {
+      localMetadata = layer.metadata
+
+      const currentLanguage = i18next.language
+
+      metadata =
+        layer.metadata &&
+        layer.metadata.metadata_id &&
+        (await getLocalMetadata(
+          localMetadataBaseUrl,
+          layer.metadata.metadata_id,
+          currentLanguage
+        ))
+      metadata = { ...metadata, title: layer.name }
+
+      //TODO: legend for external layers?
+      const layerId = layer?.id
+      const legendName =
+        localMetadata && 'legend_name' in localMetadata
+          ? localMetadata.legend_name
+          : ''
+
+      const legendHtml =
+        legendName &&
+        (await getLegendHtml(
+          legendBaseUrl,
+          legendName,
+          layerId,
+          currentLanguage
+        ))
+      metadata = legendHtml
```

Comment:
factorize this a bit better

Reply:
Indeed, I will still try to improve this with the external layers.